pkg/kclient/kclient.go

```diff
@@ -228,3 +233,38 @@ func (c *Client) IsResourceSupported(apiGroup, apiVersion, resourceName string)
 	}
 	return supported, nil
 }
+
+// IsSSASupported checks if Server Side Apply is supported by cluster
+// SSA was introduced in Kubernetes 1.16
+// If there is an error while parsing versions, it assumes that SSA is supported by cluster.
```

Comment:
I'm confused. If there's an error to parse k8s version, we assume that SSA is supported? Is this for real? Why would parsing the version info fail? And how does that lead to concluding the SSA is supported? I'm lost. 🤔

Reply:
Failing to parse an error would be an unexpected error, that should never happen.  
But that also means that it is bound to happen at least once future :-D 

Most of the k8s versions that are currently being used have support for SSA. 
So even if there is some weird version string that we can't parse odo can try to continue doing its work.
If the cluster with a weird version number supports SSA odo will be able to continue, otherwise, it would fail on a relatively meaningless version check. 
If the cluster doesn't support SSA it will fail while trying to use it.

Reply:
> Failing to parse an error would be an unexpected error, that should never happen.
> But that also means that it is bound to happen at least once future :-D

Is the k8s project going to change the versioning approach in future?

>  So even if there is some weird version string that we can't parse odo can try to continue doing its work.

Did you come across some weird version number that you could share to explain this?

My concern is that if it's an assumption then we're better off not bothering about it right now and addressing/fixing things when we hit them.

Reply:
> Is the k8s project going to change the versioning approach in future?

Don't know

> Did you come across some weird version number that you could share to explain this?

I did not.





> My concern is that if it's an assumption then we're better off not bothering about it right now and addressing/fixing things when we hit them.

Don't understand what you mean.

Reply:
Probably there's some misunderstanding, but I don't understand why we're expecting to eventually hit an error while parsing the server version.

Reply:
Resolved this over a chat with @kadel & @feloy. I was focussing more on the code part than on the SSA part.